File: Python/5.Move all negative numbers to beginning and positive to end with constant extra space.py
def separateNegativeAndPositive(nums):
    i = 0
    j = len(nums) - 1
    while i < j:
        if nums[i] > 0 and nums[j] < 0:
            nums[i], nums[j] = nums[j], nums[i]
            i += 1
            j -= 1
        elif nums[i] > 0 and nums[j] > 0:
            j -= 1
        elif nums[i] < 0 and nums[j] < 0:
            i += 1
        else:
            i += 1
            j -= 1
    return nums

File: C++/5.Move all negative numbers to beginning and positive to end with constant extra space.cpp
#include <bits/stdc++.h> 
vector<int> separateNegativeAndPositive(vector<int> &nums){
    int i = 0;
    int j = nums.size() - 1;
    while (i < j) {
        if (nums[i] > 0 && nums[j] < 0) {
            swap(nums[i], nums[j]);
            i++;
            j--;
        } else if (nums[i] > 0 && nums[j] > 0) {
            j--;
        } else if (nums[i] < 0 && nums[j] < 0) {
            i++;
        } else {
            i++;
            j--;
        }
    }
    return nums;
}
